engine/src/flutter/impeller/entity/contents/color_source_contents.h

```diff
@@ -241,19 +239,7 @@ class ColorSourceContents : public Contents {
 
     pass.SetPipeline(pipeline_callback(options));
 
-    if (!pass.Draw().ok()) {
-      return false;
-    }
-
-    // If we performed overdraw prevention, a subsection of the clip heightmap
```

Comment:
This puzzles me. Why would we have needed to restore the clip depth in the first place? Any clip that is still in effect will have a higher depth value and we wouldn't have overwritten it by our actions. It should be OK to leave stale depth values in the buffer as each rendering op is processed at a slightly higher depth value. Also, didn't the previous solution use the stencil buffer and not the depth buffer?